superset/connectors/sqla/models.py

```diff
@@ -1169,24 +1189,30 @@ def fetch_metadata(self, commit: bool = True) -> None:
         dbcols = (
             db.session.query(TableColumn)
             .filter(TableColumn.table == self)
-            .filter(or_(TableColumn.column_name == col.name for col in table.columns))
+            .filter(
+                or_(TableColumn.column_name == col.name for col in my_table.columns)
+            )
         )
         dbcols = {dbcol.column_name: dbcol for dbcol in dbcols}
 
-        for col in table.columns:
+        for col in my_table.columns:
             try:
                 datatype = db_engine_spec.column_datatype_to_string(
                     col.type, db_dialect
                 )
-            except Exception as ex:
+            except Exception as ex:  # pylint: disable=broad-except
                 datatype = "UNKNOWN"
-                logger.error("Unrecognized data type in {}.{}".format(table, col.name))
+                logger.error("Unrecognized data type in %s.%s", my_table, col.name)
                 logger.exception(ex)
             dbcol = dbcols.get(col.name, None)
             if not dbcol:
                 dbcol = TableColumn(column_name=col.name, type=datatype, table=self)
-                dbcol.sum = dbcol.is_numeric
-                dbcol.avg = dbcol.is_numeric
+                dbcol.sum = (
+                    dbcol.is_numeric
+                )  # pylint: disable=attribute-defined-outside-init
+                dbcol.avg = (
+                    dbcol.is_numeric
+                )  # pylint: disable=attribute-defined-outside-init
```

Comment:
OK, what's the deal with these two attributes? Is this cruft? Can I kill it? does anyone know?

Reply:
Yay `pylint` paying dividends. I think these could be annexed as they don't seem to be part of the `TableColumn` class nor seem to be used.

Reply:
I think that the comment applies to the block, meaning you can:
```python
# pylint: disable=attribute-defined-outside-init
dbcol.sum = dbcol.is_numeric
dbcol.avg = dbcol.is_numeric
```
https://docs.pylint.org/en/1.6.0/faq.html#is-it-possible-to-locally-disable-a-particular-message

It'd be nice to have more scoping options in `pylint`, but block is a good place to start.